### DISCUSSION TITLE
Question about re-rendering on UI change

### DISCUSSION THREAD

**Discussion by hbhutta** [vercel/next.js#58522] [Help]
### Summary

I am a beginner in web development trying to make a todo-list application. I have a button on which when I click, I want the UI to automatically re-render in such a way that the changes are reflected. For example, if I click a "delete task" button, the UI should automatically refresh to reflect those changes and quickly too. At the moment I have to manually refresh the web page in order to see the changes. 

I read the documentation and I know it's not Fast Refresh or Dynamic Rendering. What is this formally called in web-development and how can I implement it in my application?

**Comment by icyJoseph** [comment]
> At the moment I have to manually refresh the web page in order to see the changes.

Can I assume you are using Next.js, with React Server Components?

Then I guess, you are missing a `router.refresh()` call.

Otherwise, I'd like to see some code.

**Comment by nicitaacom** [comment]
Agreed with icyJoseph

Your code should look like
1. in server component you have client component DeleteTaskButton.tsx
2. in this client component you make API call and then you do `router.refresh()`

I recommend you to create notion clone first to undestand how it works
Code with Antonio have a great guide - you may find it on YouTube
